core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/stack/detach/StackDetachService.java

```diff
@@ -0,0 +1,51 @@
+package com.sequenceiq.cloudbreak.core.flow2.stack.detach;
+
+import static com.sequenceiq.cloudbreak.util.Benchmark.measure;
+
+import java.util.Date;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+
+@Service
+public class StackDetachService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackDetachService.class);
+
+    private static final String DELIMITER = "_";
+
+    @Inject
+    private ClusterService clusterService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private Clock clock;
+
+    public Stack detachStack(Stack stack) {
+        // While detaching the stack, name of the stack is renamed to avoid conflicts.
```

Comment:
There is a type in the comment. It was with the cluster name. Even the cluster name should be unique. When a new stack is created with the same name it would fail because of the existing cluster entry.